src/helpers.js

```diff
@@ -117,25 +119,85 @@ export function isTraversable(tileID, me) {
   }
 }
 
+/* eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["bigGrid"] }] */
+
+/*
+ * place all values from smallGrid into bigGrid. Align the upper left corner at x, y
+ */
+function fillGridWithSubgrid(bigGrid, smallGrid, x, y) {
+  for (let i = 0; i < smallGrid.length; i++) {
+    for (let j = 0; j < smallGrid[0].length; j++) {
+      bigGrid[i + x][j + y] = smallGrid[i][j];
+    }
+  }
+}
+
+
+/* Returns a 2d cell array of traversible (1) and blocked (0) cells inside a tile.
+ *
+ * tileTraverse: if this tile is traversable
+ * cpt: number of cells per tile
+ * objRadius: radius of untraversable object in pixels
+ */
+export function traversableCellsInTile(tileTraverse, cpt, objRadius) {
+  const gridTile = [];
+  // Start with all traversable
+  let i;
+  let j;
+  for (i = 0; i < cpt; i++) {
+    gridTile[i] = new Array(cpt);
+    for (j = 0; j < cpt; j++) {
```

Comment:
The i and j are actually reused as loop variables later, and JS forced me to not declare them with `let` twice, so I decided to declare them once at the top. I'm not sure if we can put the `let` inside the `for` in the first set next loops and then not have `let` in the second set.